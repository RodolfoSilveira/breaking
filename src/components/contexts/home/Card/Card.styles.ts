import styled from 'styled-components'

export const Wrapper = styled.div`
  background: transparent;
  width: 225px;
  margin: 40px auto 0;
  cursor: pointer;
`

export const ImageBox = styled.div`
  display: flex;
  justify-content: end;
  height: 300px;
  > img {
    height: 100%;
    width: 205px;
    object-fit: cover;
  }
`

export const ProfileBox = styled.div`
  position: relative;
  &::after {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 0;
    height: 0;
    border: 20px solid transparent;
    border-bottom-color: #acacac;
    border-top: 0;
    border-right: 0;
    margin-top: -20px;
    overflow: visible;
  }

  background-color: #fff;

  > div {
    border-top: 1px solid #000;
  }
`

export const Title = styled.p`
  margin-top: 0;
  margin-bottom: 0;
  text-align: center;
  padding: 20px 0 20px;
  font-weight: 700;
`;

export const Subtitle = styled.p`
  margin-top: 0;
  text-align: center;
  padding-bottom: 20px;
  margin-bottom: 0;
`;
