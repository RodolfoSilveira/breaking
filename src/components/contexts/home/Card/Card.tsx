import * as S from './Card.styles'

export type CardProps = {
  char_id: number
  name: string
  img: string
  nickname: string
  handleClick: (n: number) => void
}

const Card = ({ ...args }: CardProps) => (
  <S.Wrapper onClick={() => args.handleClick(args.char_id)}>
    <S.ImageBox>
      <img src={args.img} alt="profile" />
    </S.ImageBox>
    <S.ProfileBox>
      <S.Title>{args.name || 'unknown'}</S.Title>
      <S.Subtitle>{args.nickname || 'unknown'}</S.Subtitle>
    </S.ProfileBox>
  </S.Wrapper>
)

export default Card
