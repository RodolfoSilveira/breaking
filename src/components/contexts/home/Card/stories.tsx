import { Story, Meta } from '@storybook/react/types-6-0'
import Card, { CardProps } from './Card'

export default {
  title: 'Design System/Card',
  component: Card,
  parameters: {
    docs: {
      description: {
        component: 'Card',
      },
    },

  },
} as Meta

const Template: Story<CardProps> = (args) => <Card {...args} />;

export const DefaultCard = Template.bind({});

DefaultCard.args = {
  name: 'Group Leader',
  nickname: 'Counselor',
  img: "https://vignette.wikia.nocookie.net/breakingbad/images/8/81/Group_Leader_Brown_Shirt_-_Kafkaesque.png/revision/latest?cb=20131025095333",
}
