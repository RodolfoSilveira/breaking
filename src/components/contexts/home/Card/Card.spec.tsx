/* eslint-disable @typescript-eslint/no-empty-function */
import { screen, render } from '@testing-library/react';
import Card from '.'

describe('<Card />', () => {
  it('should render a card', () => {
    const { container } = render(<Card
      name="teste"
      nickname=""
      img="teste"
      handleClick={() => {}}
      char_id={1}
    />)

    expect(screen.getByText(/unknown/i)).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
});
