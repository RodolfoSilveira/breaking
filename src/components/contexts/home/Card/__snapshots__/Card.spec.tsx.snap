// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render a card 1`] = `
.c0 {
  background: transparent;
  width: 225px;
  margin: 40px auto 0;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  height: 300px;
}

.c1 > img {
  height: 100%;
  width: 205px;
  object-fit: cover;
}

.c2 {
  position: relative;
  background-color: #fff;
}

.c2::after {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 0;
  border: 20px solid transparent;
  border-bottom-color: #acacac;
  border-top: 0;
  border-right: 0;
  margin-top: -20px;
  overflow: visible;
}

.c2 > div {
  border-top: 1px solid #000;
}

.c3 {
  margin-top: 0;
  margin-bottom: 0;
  text-align: center;
  padding: 20px 0 20px;
  font-weight: 700;
}

.c4 {
  margin-top: 0;
  text-align: center;
  padding-bottom: 20px;
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="profile"
        src="teste"
      />
    </div>
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        teste
      </p>
      <p
        class="c4"
      >
        unknown
      </p>
    </div>
  </div>
</div>
`;
